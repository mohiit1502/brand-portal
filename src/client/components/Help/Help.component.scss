.c-Help {
    &.container {
        margin-left: unset;
        margin-right: unset;
        max-width: unset;
    }

    .c-Help__header {
        padding: 1.5rem 2.5rem;
        background: #FAFAFA;
    }

    .c-Help__content {
        padding: 3rem 2.5rem 1.5rem 2rem;
    }
}
