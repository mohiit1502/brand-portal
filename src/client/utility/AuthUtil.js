import authConfig from "./../config/authorizations";
import Helper from "./helper";

export default class AuthUtil {
  static isActionAccessible (action, userProfile, actedOnRole) {
    if (action && action.indexOf(".") > -1) {
      const actionConfig = Helper.search(action, authConfig);
      if (actionConfig && userProfile && userProfile.role && userProfile.role.name && typeof actionConfig === "object") {
        const role = userProfile.role.name;
        if (actionConfig.length !== undefined) {
          return actionConfig.includes(role);
        } else {
          if (Object.keys(actionConfig).includes(role) && actedOnRole && actionConfig[role].includes(actedOnRole)) {
            return true;
          }
        }
      }
    }
    return false;
  }
}
