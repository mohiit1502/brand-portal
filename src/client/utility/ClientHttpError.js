export default class ClientHttpError {

  status;
  error;
  message;

  constructor (status, error, message) {
    this.status = status;
    this.error = error;
    this.message = message;
  }

}
