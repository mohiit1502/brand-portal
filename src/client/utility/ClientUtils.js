import queryString from "query-string";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }
}
