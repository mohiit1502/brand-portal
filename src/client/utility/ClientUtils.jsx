import React from "react";
import queryString from "query-string";
import {Redirect} from "react-router";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }

  static  getAllValuesFromRecursiveTree(tree) {

    if (typeof tree !== "object" || tree === null) {
      return [tree];
    }

    const list = [];

    for (const i in tree) {
      if (tree.hasOwnProperty(i)) {
        list.push(...this.getAllValuesFromRecursiveTree(tree[i]));
      }
    }
    return list;
  }

  static where (arrayList, filterObj) {
    let index = -1;
    for (let i = 0; i < arrayList.length; i++) {
      for (const j in filterObj) {
        if (filterObj.hasOwnProperty(j) && arrayList[i].hasOwnProperty(j) && filterObj[j] === arrayList[i][j]) {
          index = i;
        } else {
          index = -1;
        }
      }
      if (index !== -1) {
        return index;
      }
    }
    return index;
  }

}
