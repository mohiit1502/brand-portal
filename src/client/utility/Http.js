/* eslint-disable complexity */
/* eslint-disable no-unused-expressions */
/* eslint-disable max-params */
/* eslint-disable max-statements */
import fetch from "node-fetch";
import queryString from "query-string";
import ClientHttpError from "./ClientHttpError";
import { NOTIFICATION_TYPE } from "../actions/notification/notification-actions";
import CONSTANTS from "../constants/constants";
export default class Http {

  static async get(url, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});

    const options = {
      method: "GET"
    };
    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;

    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message, err.code);
  }

  static async post(url, data, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "POST",
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json"
      }
    };
    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;
    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message);

  }

  static async postAsFormData(url, data, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "POST",
      body: data,
      headers: { }
    };
    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;
    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message);

  }

  static async put(url, data, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "PUT",
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json"
      }
    };

    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;
    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message);
  }

  static async delete(url, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "DELETE",
      noInject: true
    };
    return fetch(urlString, options);
  }

  displayToast(status, toastcallback) {
    if (new RegExp(CONSTANTS.CODES.ERRORCODES.SERVERERROR).test(status)) {
      toastcallback && typeof toastcallback === "function" && toastcallback(NOTIFICATION_TYPE.ERROR, `Unable to reach our services currently, please try again in sometime.`);
    } else if (new RegExp(CONSTANTS.CODES.ERRORCODES.FOURNOTFOUR).test(status)) {
      toastcallback && typeof toastcallback === "function" && toastcallback(NOTIFICATION_TYPE.ERROR, `Requested Resource could not found.`);
    }
  }
}
