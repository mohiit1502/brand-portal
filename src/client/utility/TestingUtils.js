import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../reducers";
import thunk from "redux-thunk";
import Immutable from "immutable";
import configureMockStore from 'redux-mock-store';
import ClientHttpError from "./ClientHttpError";

export const findByTestAttribute = (component,attr) => component.find(`[data-test="${attr}"]`);

export const testStore = state => configureMockStore()(state);

export const realStore = state => {
  const configureStore = initialState => {
    const composeEnhancers =
      typeof window === "object" &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
          serialize: { // prettier-ignore
            immutable: Immutable
          }
        }) : compose;
    const store = createStore(rootReducer, initialState, composeEnhancers(applyMiddleware(thunk)));

    if (module.hot) {
      module.hot.accept("./reducers", () => {
        const nextRootReducer = require("./reducers").default;
        store.replaceReducer(nextRootReducer);
      });
    }

    return store;
  };

  return configureStore(state || window.__WML_REDUX_INITIAL_STATE__);
}

export const clearKeys = (tree, arr) => {
  if (arr && arr.indexOf(tree) > -1) {
    return;
  }
  if (tree && typeof tree === 'object') {
    if (tree.inputId) {
      return;
    }
    arr.push(tree)
    if (tree.key) {
      try {
        delete tree.key;
      } catch (e) {}
    }
    if (Array.isArray(tree)) {
      tree.forEach(obj => clearKeys(obj, arr));
    } else {
      Object.keys(tree).forEach(key => clearKeys(tree[key], arr));
    }
  }
}

export function setupFetchStub(data) {
  return function fetchStub(_url) {
    return new Promise((resolve) => {
      resolve({
        json: () =>
          Promise.resolve({
            data,
          }),
      })
    })
  }
}

export function setupFetchThrowStub() {
  const error = new ClientHttpError(500, "test-error"); // Directly throwing new ClientHttpError fails in Jest
  throw error;
}
