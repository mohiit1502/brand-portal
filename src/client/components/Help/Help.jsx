/* eslint-disable filenames/match-regex */
import React, {useEffect, useState} from "react";
import PropTypes from "prop-types";
import {withRouter} from "react-router";
import HelpMain from "./../HelpMain";
import HelpSideBar from "./../HelpSideBar";
import ImageViewer from "../ImageViewer/ImageViewer";
import Http from "./../../../client/utility/Http";
import helpConfiguration from "../../config/contentDescriptors/help";
import "./Help.component.scss";

const Help = props => {

  const [activeTabUrl, setActiveTabUrl] = useState(props.location.pathname);
  const [helpConfig, setHelpConfig] = useState(helpConfiguration);
  const [loader, setLoader] = useState(true);
  useEffect(() => {
    (async () => {
      setLoader(false);
      try {
        setHelpConfig(helpConfiguration);
        const response = (await Http.get("/api/helpConfig")).body;
        setHelpConfig(JSON.parse(response));
      /* eslint-disable no-empty */
      } catch (e) {}
    })();
  }, []);

  useEffect(() => {
    setActiveTabUrl(props.location.pathname);
  }, [props.location.pathname]);

  return (
    <div className={`${loader ? " loader" : ""}`}>
      { helpConfig && Object.keys(helpConfig).length > 0 &&
        <div className="c-Help container h-100">
          <div className="c-Help__header row">
            <div className="col-12">
              <div className="h3 help-header">{helpConfig.header}</div>
            </div>
          </div>
          <div className="c-Help__content row h-100">
            <div className="col-2 h-100">
              <HelpSideBar categoryHeader={helpConfig.categoryHeader} categories={helpConfig.categories}
                activeTab={helpConfig.sectionMap[activeTabUrl]}/>
            </div>
            <div className="col-10">
              <HelpMain content={helpConfig.content} activeTab={helpConfig.sectionMap[activeTabUrl]}/>
            </div>
          </div>
          <ImageViewer />
        </div>
      }
    </div>
    );
};

Help.propTypes = {
  location: PropTypes.object
};

export default withRouter(Help);
