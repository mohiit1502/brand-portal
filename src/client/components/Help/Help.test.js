/* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
import React, {useRef} from "react";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import {testStore} from "../../utility/TestingUtils";
import Help from "./";
import helpData from "../../../../test/client/mocks/helpData";
import Http from "../../utility/Http";

let store;

const setUp = () => {
  store = testStore({});
  return renderer.create(<Provider store={store}><BrowserRouter><Help /></BrowserRouter></Provider>);
};

jest.mock("react", () => {
  const originReact = jest.requireActual("react");
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef
  };
});

describe("Help", () => {
  let wrapper;
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetAllMocks();
  });
  it("should render help successfully", () => {
    jest.spyOn(Http, "get").mockImplementation(() => Promise.resolve({body: helpData}));
    const mRef = {current: document.createElement("div")};
    useRef.mockReturnValue(mRef);
    wrapper = setUp();
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
