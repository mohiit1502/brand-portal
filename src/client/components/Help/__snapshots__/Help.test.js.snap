// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help should render help successfully 1`] = `
<div
  className=" loader"
/>
`;
